py-scripts/lf_ftp.py

```diff
@@ -1659,20 +1848,104 @@ def generate_report(self, ftp_data, date, input_setup_info, test_rig, test_tag,
         self.report.set_table_title("Overall Results")
         self.report.build_table_title()
         # self.report.test_setup_table(value="Information", test_setup_data=input_setup_info)
-        dataframe = {
-            " Clients": client_list,
-            " MAC ": self.mac_id_list,
-            " Channel": self.channel_list,
-            " SSID ": self.ssid_list,
-            " Mode": self.mode_list,
-            " No of times File downloaded ": self.url_data,
-            " Time Taken to Download file (ms)": self.uc_avg,
-            " Bytes-rd (Mega Bytes)": self.bytes_rd,
-            "RX RATE (Mbps)": self.rx_rate
-        }
-        dataframe1 = pd.DataFrame(dataframe)
-        self.report.set_table_dataframe(dataframe1)
-        self.report.build_table()
+        if (self.clients_type == 'Real'):
+            # Calculating the pass/fail criteria when either expected_passfail_val or csv_name is provided
+            if (self.expected_passfail_val is not None or self.csv_name is not None):
+                # When csv_name is provided, for pass/fail criteria, respective values for each client will be used
+                if (self.expected_passfail_val == '' or self.expected_passfail_val is None):
+                    res_list = []
```

Comment:
Please move the pass/fail logic to a separate helper function. This should really belong in it's own commit, as we've discussed previously. While I will not reject this commit outright for this in this bunch of PRs, I will be more stringent on this point moving forward, given we've discussed this now multiple times.